chore(blog): 10 years of electron (#382)
electron/website: blog/10-years-of-electron.md

slug: 10-years-of-electron
---

# 10 Years of Electron 🎉

The first commit to the `electron/electron` repository was on March 13, 2013[^1].

![Initial commit on electron/electron by @aroben](/assets/img/first-commit.png)

10 years and 27,147 more commits from 1192 unique contributors later, Electron has become one of the
most popular frameworks for building desktop applications today. This milestone is the perfect
opportunity to celebrate and reflect on our journey so far, and to share what we’ve learned along
the way.

We would not be here today without everyone who has dedicated their time and effort to contribute to
the project. Although source code commits are always the most visible contributions, we also have to
acknowledge the effort of folks who report bugs, maintain userland modules, provide documentation
and translations, and participate in the Electron community across cyberspace.
Every contribution is invaluable to us as maintainers.

**Before we continue with the rest of the blog post: thank you. ❤️**

## How did we get here?

**Atom Shell** was built as the backbone for GitHub’s [Atom editor](https://atom.io/), which launched in

> public beta in April 2014. It was built from the ground-up as an alternative to the web-based

Reviewer: ```suggestion
ground-up
```

desktop frameworks available at the time (node-webkit and Chrome Embedded Framework). It had a
killer feature: embedding Node.js and Chromium to provide a powerful desktop runtime for web
technologies.

Within a year, Atom Shell began seeing immense growth in capabilities and popularity.
Large companies, startups, and individual developers alike had started building apps with it
(some early adopters include [Slack](https://slack.com), [GitKraken](https://gitkraken.com), and
[WebTorrent](https://webtorrent.io)), and the project was aptly renamed to **Electron**.

From then on, Electron hit the ground running and never stopped. Here’s a look at our weekly
download count over time, courtesy of [npmtrends.com](https://npmtrends.com/electron):

![Electron weekly downloads graph over time](/assets/img/weekly-download-graph.png)

Electron v1.0 was released in 2016, promising increased API stability and better docs and tooling.

> Electron v2.0 released in 2018 and introduced semantic versioning, making it easier for Electron

Reviewer: ```suggestion
Electron v2.0 was released in 2018 and introduced semantic versioning, making it easier for Electron
```

developers to keep track of the release cycle.

By Electron v6.0, we shifted to a regular 12-week major release cadence to match Chromium’s. This
decision was a change in mentality for the project, bringing “having the most up-to-date Chromium
version” from a nice-to-have to a priority. This has reduced the amount of tech debt between
upgrades, making it easier for us to keep Electron updated and secure.

Since then, we’ve been a well-oiled machine, releasing a new Electron version on the same day as
every Chromium stable. By the time Chromium sped up their release schedule to 4 weeks in 2021,
we were able to shrug our shoulders and increase our release cadence to 8 weeks accordingly.

We’re now on Electron v23.0 (and counting), and are still dedicated to building the best runtime for
building cross-platform desktop applications. Even with the boom in JavaScript developer tools in
recent years, Electron has remained a stable, battle-tested stalwart of the desktop app framework
landscape. Electron apps are ubiquitous nowadays: you can program with Visual Studio Code, design
with Figma, communicate with Slack, and take notes with Notion (amongst many other use cases).

> We’re incredibly proud of this achievement and grateful to everyone who has made it possible.

Reviewer: Potentially worth switching either vX _or_ vX.X.X in all instances, since that's generally how the ecosystem looks at versions.

Reviewer 2: minor/patch probably doesn't matter here, e.g. "We're now on Electron v23" and "By Electron v6, we shifted to a regular 12-week major release cadence" flows better.

Author: [238eadf](https://github.com/electron/website/pull/382/commits/238eadfe656329205417f2d49e711ad1fb096378)

- In 2021, we established the [Electron China](https://github.com/electronjs-cn) user group with the help of
  [@BlackHole1](https://github.com/BlackHole1). This group has been instrumental in Electron growth
  in users from China’s booming tech scene, providing a space for them to collaborate on ideas and
  discuss Electron outside of our English-language spaces. We’d also like to thank
  [cnpm](https://npmmirror.com/) for their work in supporting Electron’s nightly releases in their
  Chinese mirror for npm.

#### Participating in high-visibility open source programs

- We have been celebrating [Hacktoberfest](https://hacktoberfest.com/) every year since 2019.
  Hacktoberfest is yearly celebration of open source organized by DigitalOcean, and we get dozens of
  enthusiastic contributors every year looking to make their mark on open source software.
- In 2020, we participated in the initial iteration of Google Season of Docs, where we worked with
  [@bandantonio](https://github.com/bandantonio) to rework Electron’s new user tutorial flow.
- In 2022, we mentored a Google Summer of Code student for the first time.

>   [@aryanshridhar](https://github.com/aryanshridhar) did some awesome work to refactor Electron
>   Fiddle’s core version loading logic and migrate its bundler from Parcel 1 to webpack.

Reviewer 2: ```suggestion
  [@aryanshridhar](https://github.com/aryanshridhar) did some awesome work to refactor [Electron
  Fiddle](https://github.com/electron/fiddle)'s core version loading logic and migrate its bundler 
  to [webpack](https://webpack.js.org/).
```

### Automate all the things!

Today, Electron governance has about 30 active maintainers. Less than half of us are full-time

> contributors, which means that there’s a lot of work to go around. What’s our trick to keep

Reviewer 2: ```suggestion
contributors, which means that there’s a lot of work to go around. What’s our trick to keeping
```

(I think?)

we built out a suite of smaller bots that help do the dirty work for us. Here are a
few examples:

- [Sudowoodo](https://github.com/apps/sudowoodo-release-bot) automates the Electron release process
  from start to finish, from kicking off builds to uploading the release assets to GitHub and npm.
- [Trop](https://github.com/electron/trop) automates the backporting process for Electron by
  attempting to cherry-pick patches to previous release branches based on GitHub PR labels.
- [Roller](https://github.com/electron/roller) automates rolling upgrades of Electron’s Chromium
  and Node.js dependencies.
- [Cation](https://github.com/electron/cation) is our status check bot for electron/electron PRs.

Altogether, our little family of bots has given us a huge boost in developer productivity!

#### What’s next?

As we enter our second decade as a project, you might be asking: what’s next for Electron?

Not much is changing in the immediate future. We’re going to keep releasing major versions of
Electron regularly every 8 weeks, keeping the framework updated with the latest and greatest from
the web platform and Node.js while maintaining stability and security for enterprise-grade

> applications.

Reviewer 3: I agree with @ckerr, people will 100% quote you on this. I _do_ like the message of consistency, that's what people building apps really want.

> Aside from that, we generally announce news on upcoming initiatives when they become concrete.
> If you want to keep up with future releases, features, and general project updates, you can read
> [our blog](https://electronjs.org/blog) and follow our social media profiles
> ([Twitter](https://twitter.com/electronjs) and [Mastodon](https://social.lfx.dev/@electronjs))!

Reviewer 2: ```suggestion
We generally announce news on upcoming initiatives when they become concrete. If you want to 
keep up with future releases, features, and general project updates, you can read 
[our blog](https://electronjs.org/blog) and follow our social media profiles 
([Twitter](https://twitter.com/electronjs) and [Mastodon](https://social.lfx.dev/@electronjs))!
```
